content/en/docs/languages/go/generated-code.md

```diff
@@ -14,6 +14,103 @@ to be run on concurrent goroutines. But also note that for <i>individual streams
+
+In this context, [`grpc.BidiStreamingServer`](https://pkg.go.dev/google.golang.org/grpc#BidiStreamingServer) can be used to access both the client-to-server message stream and the server-to-client message stream. 
+
+## Methods on generated client interfaces
+
+For client side usage, each `service Bar` in the `.proto` file also results in the function: `func BarClient(cc *grpc.ClientConn) BarClient`, which
+returns a concrete implementation of the `BarClient` interface (this concrete implementation also lives in the generated `.pb.go` file).
+
+### Unary Methods
+
+These methods have the following signature on the generated client stub:
+
+`(ctx context.Context, in *MsgA, opts ...grpc.CallOption) (*MsgB, error)`
+
+In this context, `MsgA` is the single request from client to server, and `MsgB` contains the response sent back from the server.
+
+### Server-Streaming methods
+
+These methods have the following signature on the generated client stub:
+
+`Foo(ctx context.Context, in *MsgA, opts ...grpc.CallOption) (grpc.ServerStreamingClient[*MsgB], error)`
+
+In this context, [`grpc.ServerStreamingClient`](https://pkg.go.dev/google.golang.org/grpc#ServerStreamingClient) represents the client side of server-to-client stream of `MsgB` messages.
+
+### Client-Streaming methods
+
+These methods have the following signature on the generated client stub:
+
+`Foo(ctx context.Context, opts ...grpc.CallOption) (grpc.ClientStreamingClient[*MsgA, *MsgB], error)`
+
+In this context, [`grpc.ClientStreamingClient`](https://pkg.go.dev/google.golang.org/grpc#ClientStreamingClient) represents the client side of client-to-server stream of `MsgA` messages. It can be used both to send the client-to-server message stream and to receive the single server response message.
+
+### Bidi-Streaming methods
+
+These methods have the following signature on the generated client stub:
+
+`Foo(ctx context.Context, opts ...grpc.CallOption) (grpc.BidiStreamingClient[*MsgA, *MsgB], error)`
+
+In this context, [`grpc.BidiStreamingClient`](https://pkg.go.dev/google.golang.org/grpc#BidiStreamingClient) represents both the client-to-server and server-to-client message streams.
+
+## Packages and Namespaces
+
+When the `protoc` compiler is invoked with `--go_out=plugins=grpc:`, the `proto package` to Go package translation
+works the same as when the `protoc-gen-go` plugin is used without the `grpc` plugin.
+
+So, for example, if `foo.proto` declares itself to be in `package foo`, then the generated `foo.pb.go` file will also be in
+the Go `package foo`.
+
+For reference to the generated code that does not use generic interfaces for streams see below.
+
+# Generated-code(non-generic) reference
+
+This describes the code generated with the [grpc plugin](https://pkg.go.dev/google.golang.org/grpc/cmd/protoc-gen-go-grpc), `protoc-gen-go-grpc`,
+when compiling `.proto` files with `protoc` without the generics.
+
+
 ## Methods on generated server interfaces
```

Comment:
Can we move the old documentation into another page and link to it from the top of this page?  Just a little note like `"The following documents the latest generated code guide which uses generics.  For documentation about the older version of generated code that did not use generics, please see <this document>."`?

Reply:
Done.